pandas/core/dtypes/cast.py

```diff
@@ -1505,12 +1506,15 @@ def cast_scalar_to_array(shape, value, dtype: Optional[DtypeObj] = None) -> np.n
 
     """
     if dtype is None:
-        dtype, fill_value = infer_dtype_from_scalar(value)
+        dtype, fill_value = infer_dtype_from_scalar(value, pandas_dtype=True)
```

Comment:
glancing through `infer_dtype_from_scalar`, it doesn't look like this will work for 3rd party extension types.

That's probably an OK limitation for now, but we'll need to address it some day. It's complicated by the fact that it might be ambiguous which extension type a scalar belongs to (we might have multiple extension arrays that could hold, e.g. integers).

Reply:
I see, I can add a comment about this as a future TODO?
```suggestion
        # TODO: Update this function to add support for 3rd party extension types
        dtype, fill_value = infer_dtype_from_scalar(value, pandas_dtype=True)
```

Reply:
Yeah, can you open an issue and link to it in the todo? You can use one of the example extension arrays in `pandas/tests/extension` like `DecimalArray` to demonstrate.

Reply:
Done, added the comment and created [Issue 34959](https://github.com/pandas-dev/pandas/issues/34959)